eclair-core/src/main/scala/fr/acinq/eclair/channel/Channel.scala

```diff
@@ -722,97 +722,97 @@ class Channel(val nodeParams: NodeParams, val wallet: OnChainChannelFunder, remo
    */
 
   when(NORMAL)(handleExceptions {
-    case Event(c: CMD_ADD_HTLC, d: DATA_NORMAL) if d.localShutdown.isDefined || d.remoteShutdown.isDefined =>
+    case Event(c: CMD_ADD_HTLC, DATA_NORMAL(d)) if d.localShutdown.isDefined || d.remoteShutdown.isDefined =>
       // note: spec would allow us to keep sending new htlcs after having received their shutdown (and not sent ours)
       // but we want to converge as fast as possible and they would probably not route them anyway
       val error = NoMoreHtlcsClosingInProgress(d.channelId)
       handleAddHtlcCommandError(c, error, Some(d.channelUpdate))
 
-    case Event(c: CMD_ADD_HTLC, d: DATA_NORMAL) =>
+    case Event(c: CMD_ADD_HTLC, DATA_NORMAL(d)) =>
```

Comment:
I'd rather use:
```suggestion
    case Event(c: CMD_ADD_HTLC, normal: DATA_NORMAL) =>
```
otherwise the pattern matching breaks every time we change the case class args. We can always `import normal._` to avoid `normal.data` in the body (but I think `normal.data` looks great).

Reply:
Done in https://github.com/ACINQ/eclair/pull/2173/commits/68d479fa9b720c40aa29e93ee669f23b51005007, don't hesitate to nitpick more on those, since I'm touching all those lines already, it's worth making it as good as we can!

Be careful when reviewing though, I hope I didn't mess something up without noticing.